src/coreclr/jit/lsrabuild.cpp

```diff
@@ -3846,9 +3846,23 @@ int LinearScan::BuildDelayFreeUses(GenTree*         node,
             return 0;
         }
     }
+
+    // Don't mark as delay free if there is a mismatch in register types
+    bool addDelayFreeUses = false;
+    // Multi register nodes should no go via this route.
+    assert(!node->IsMultiRegNode());
+    if (rmwNode == nullptr || varTypeUsesSameRegType(rmwNode->TypeGet(), node->TypeGet()) ||
+        (rmwNode->IsMultiRegNode() && varTypeUsesFloatReg(node->TypeGet())))
```

Comment:
did not understand the `&& varTypeUsesFloatReg(node->TypeGet())` here? did you see code where having delay free is ok to have for multi-reg node that are GPR (if there is such a thing)?

Reply:
If the RMW node is a multi register, then it'll always be vector registers.

The normal node could be general register or vector register. We want to discount the case where node is a general register. So, `varTypeUsesFloatReg()` is confirming `node` uses a FP, vector or sve vector register

Eg:
```
(Vector128<sbyte> Value1, Vector128<sbyte> Value2) LoadAndInsertScalar((Vector128<sbyte>, Vector128<sbyte>) values, [ConstantExpected(Max = (byte)(15))] byte index, sbyte* address)
```

The RMW node is the multi register `values`.
This code makes sure `index` is not delay slotted.

Reply:
I am wondering we should have an assert that if it is a multiRegNode, then `node` should be `floatreg`?

Reply:
Added some asserts.

That then broke things because for `delayFreeMultiple` intrinsics we do:
```
BuildDelayFreeUses(use.GetNode(), intrinsicTree);
```
Which is passing the entire `intrinsicTree` down as the `rmw` node which feels wrong. Fixed that to pass `op1` instead.